### PR TITLE
Refactoring and expanding the Media section

### PR DESCRIPTION
**Pull request by derekherman**
We’re proposing to make [web.dev/media](https://web.dev/media) the single place for prescriptive guidance on how to build great media experiences. The site will tell developers what makes a great media experience, describe various tradeoffs, and provide strong recommendations where applicable. The content changes will happen in multiple phases with the initial phase being this pull request. In phase one we're focusing on reorganizing the various sections and updating the existing content to ensure a smooth transition between articles and consistent voice that builds with each progression through the various sections.

The “Concepts” section should be high-level describing all the foundational knowledge required to read the other two sections and be more about the theories i.e. concepts. The “Prepare media files for the web” section should build on the “Concepts” section and dive deeper into those topics with references and examples, and additionally move some of the “Concepts” content into this section while breaking up the cheat sheet into separate pages. The final section “Add media to a web page” should focus on doing, and dive much deeper into the previous topics. This is the section where we will add new content that describes the demo PWA that was built as part of this content revamp.

All of the pages have been read by multiple people and looked at for ways to improve the flow and build on what was already there. The majority of the original content is still intact with either a little manipulating or enhancement to connect it to the new structure and added pages. Every single page in the media section has been in some way edited. This PR is very extensive and represents the first phase as we continue to write more content that will dive deeper into areas like DRM, WebRTC, Picture-in-picture and many more.

Again the focus here is foundational and mainly on streaming video (VOD) and offline playback in relation to the demo PWA, while also diving a little bit deeper into all of the existing content including a lot more practical examples on how to manipulate video files with Shaka Packager and even a demo Docker file that makes it much easier to install for the average person. Phase two will be mostly new original pages and be much less focused on updating what is already there.

Tracking changes proposed in this pull request:

- [x] Add new authors
- [x] Provide author images
- [x] Update URLs referencing the new demo PWA, and repo, once it's been deployed to production
- [x] Restructure `/media` section content & page orders
- [x] Refactor `/media-experience`
- [x] Refactor `/media-file-basics`
- [x] Refactor `/media-application-basics`
- [x] Add `/media-streaming-basics`
- [x] Refactor `/prepare-media`
- [x] Refactor `/containers-and-codecs`
- [x] Refactor `/bitrate`
- [x] Refactor `/resolution`
- [x] Refactor `/media-cheat-sheet` and rename/redirect to`/media-conversion`
- [x] Add `/media-encryption`, include refactored content from `/media-cheat-sheet`
- [x] Refactor `/add-media`
- [x] Refactor `/video-and-source-tags`
- [x] Refactor `/media-accessibility`
- [x] Add `/media-frameworks`
- [x] Add `/pwa-with-offline-streaming`

### CODE REVIEW

**Comment by derekherman** [comment]
@googlebot I consent.

**Comment by dero** [comment]
@googlebot I consent.

File: src/site/content/en/media/video-and-source-tags/index.md [hunk 1]
```diff
@@ -4,36 +4,34 @@ title: The <video> and <source> tags
...
-For anyone to see it, you still need to add it to a web page. Doing so properly
-requires two HTML elements: the
-[`<video>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video)
-element and the
-[`<source>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/source).
-In addition to basics about these tags, this section also explains attributes
-you should add to those tags to craft a good user experience.
+For anyone to see your video, you still need to add it to a web page. Doing so
+properly requires adding two HTML elements: the [`<video>`] element and the
+[`<source>`] to your web page. In addition to basics about these tags, this
```
Comment (jpmedley) [inline src/site/content/en/media/video-and-source-tags/index.md:26]:
```suggestion
[`<source>`] element. In addition to basics about these tags, this
```

File: src/site/content/en/media/video-and-source-tags/index.md [hunk 2]
```diff
@@ -4,36 +4,34 @@ title: The <video> and <source> tags
...
+article also explains attributes you should add to those tags to craft a good
```
Comment (jpmedley) [inline src/site/content/en/media/video-and-source-tags/index.md:27]:
```suggestion
article explains attributes you should add to those tags to craft a good
```

File: src/site/content/en/media/video-and-source-tags/index.md [hunk 3]
```diff
@@ -64,7 +63,13 @@ The example below produces the embedded video that immediately follows.
 </video>
 ```
 
-[Try it](https://track-demonstration.glitch.me) ([source](https://glitch.com/edit/#!/track-demonstration))
+[Try it on Glitch] ([source])
+
+{% Aside %}
+Notice in the previous example that the `controls` attribute was introduced.
+which instructs browsers to allow the user to control video playback, including
```
Comment (jpmedley) [inline src/site/content/en/media/video-and-source-tags/index.md:70]:
```suggestion
This instructs browsers to allow the user to control video playback, including
```

File: src/site/content/en/media/video-and-source-tags/index.md [hunk 4]
```diff
@@ -249,20 +247,23 @@ different from the original video. Squashed or stretched videos looks bad.
...
 </div>
 ```
 
-[Try it](https://googlesamples.github.io/web-fundamentals/fundamentals/media/responsive_embed.html)
+[Try it]
 
-Compare the [responsive sample](https://googlesamples.github.io/web-fundamentals/fundamentals/media/responsive_embed.html)
-to the [unresponsive version](https://googlesamples.github.io/web-fundamentals/fundamentals/design-and-ux/responsive/unyt.html).
+Compare the [responsive sample] to the [unresponsive version]. As you can see,
+it's not a great user experience with the unresponsive version.
```
Comment (jpmedley) [inline src/site/content/en/media/video-and-source-tags/index.md:266]:
```suggestion
the unresponsive version isn't a great user experience.
```

File: src/site/content/en/media/video-and-source-tags/index.md [hunk 5]
```diff
@@ -303,20 +303,19 @@ The `autoplay` attribute controls whether the browser downloads and plays a
 video immediately. The precise way it works depends on the platform and browser.
 
 * Chrome: Depends on multiple factors including but not limited to whether the
-  viewing is on desktop and whether the a mobile user has added your site or app
-  to their homescreen. For  details, see [Autoplay Policy
-  Changes](https://developers.google.com/web/updates/2017/09/autoplay-policy-changes).
+  viewing is on desktop and whether the mobile user has added your site or app
+  to their homescreen. For details, see [Autoplay best practices].
 
 * Firefox: Blocks all video and sound, but gives user the ability to relax these
```
Comment (jpmedley) [inline src/site/content/en/media/video-and-source-tags/index.md:309]:
```suggestion
* Firefox: Blocks all video and sound, but gives users the ability to relax these
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: src/site/content/en/media/bitrate/index.md [hunk 6]
```diff
@@ -48,20 +50,39 @@ doing your own testing.
...
 least: desktop, 4G, and 3G. If you're serving video in one of the markets
 referred to as "the next billion users", say India, for example, you'll want to
-include 2G as well. For demonstration purposes, I'm going to target 3G.
+include 2G as well. For demonstration purposes, we're going to target 3G.
 
-In FFmpeg you set the bitrate with the (surprise!) bitrate (`-b`) flag.
+Using FFmpeg you set the bitrate with the (surprise!) bitrate (`-b`) flag.
+
+If you don't have FFmpeg installed read
+[Media application basics](/media-application-basics/#installing-applications-with-docker)
```
[the remark this answers is not in the document]
Comment (robdodson) [reply src/site/content/en/media/bitrate/index.md:63]: Not that I know of. It's free open source software so it should be ok for folks to use.

File: src/site/content/en/media/add-media/index.md [hunk 7]
```diff
@@ -3,32 +3,47 @@ layout: post
...
+  we built an updated demo PWA that is capable of adaptive streaming and offline
+  playback—without using frameworks, plus you can play with source code.
 
 {% Aside %}
 Be sure not to skip the accessibility section. The technical aspects of
 supporting accessibility are not that difficult. They are also a regulatory or
 legal requirement in many places.
 {% endAside %}
+
+There is a lot of ground to cover in this section, let's get started by learning
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/add-media/index.md:48]: @jpmedley Do you want to just remove the `let's` from the sentence?

File: src/site/content/en/media/bitrate/index.md [hunk 8]
```diff
@@ -3,41 +3,43 @@ layout: post
...
   - media
   - video
 ---
 
-In the [previous section](/containers-and-codecs), I showed you how to change
-a media file's container and codec. In this section, I show how to change
-bitrate before explaining [resolution](/resolution) and, finally, [how to
-embed your file in a page](/add-media).
+In the previous [Containers and codecs](/containers-and-codecs/) article, we
+learned how to change the extension (container) and codec of a media file. In
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/bitrate/index.md:18]: That makes sense to me. I think swapping them would read better.

File: src/site/content/en/media/bitrate/index.md [hunk 9]
```diff
@@ -3,41 +3,43 @@ layout: post
...
 tags:
   - media
   - video
 ---
 
-In the [previous section](/containers-and-codecs), I showed you how to change
-a media file's container and codec. In this section, I show how to change
-bitrate before explaining [resolution](/resolution) and, finally, [how to
-embed your file in a page](/add-media).
+In the previous [Containers and codecs](/containers-and-codecs/) article, we
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/bitrate/index.md:17]: Good catch

File: src/site/content/en/media/pwa-with-offline-streaming/index.md [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+  };
+  event.respondWith(getResponse());
+};
+self.addEventListener('fetch', fetchHandler);
+```
+
+So what do you need to do in `getVideoResponse`?
+
+- The `event.respondWith` method expects a `Response` object as a parameter.
+- The [Response constructor] tells us that there are several types of objects we
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/pwa-with-offline-streaming/index.md:244]: This change means we also need to change the Markdown link at the bottom of the page to reflect the update text.

File: src/site/content/en/media/pwa-with-offline-streaming/index.md [hunk 11]
```diff
@@ -0,0 +1,338 @@
...
+    return fetch(event.request);
+  };
+  event.respondWith(getResponse());
+};
+self.addEventListener('fetch', fetchHandler);
+```
+
+So what do you need to do in `getVideoResponse`?
+
+- The `event.respondWith` method expects a `Response` object as a parameter.
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/pwa-with-offline-streaming/index.md:243]: @jpmedley not sure what this change is for. Why add `90` to the string?

File: src/site/content/en/media/pwa-with-offline-streaming/index.md [hunk 12]
```diff
@@ -0,0 +1,338 @@
...
+
+## Downloading and storing a large media file
+
+Progressive Web Apps usually use the convenient [Cache API] to both download
+and store the assets required to provide the offline experience: documents,
+stylesheets, images and others.
+
+Here is a basic example of using the Cache API within a Service Worker:
+
+```javascript
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/pwa-with-offline-streaming/index.md:50]: Shouldn't we also remove the space then? By my count the lines being highlighted don't line up with the file.

File: src/site/content/en/media/pwa-with-offline-streaming/index.md [hunk 13]
```diff
@@ -0,0 +1,338 @@
...
+  - video
+---
+
+[Progressive Web Apps] bring a lot of features previously reserved for native
+applications to the web. One of the most prominent features associated with
+PWAs is an offline experience.
+
+Even better would be an offline streaming media experience, which is an
+enhancement you could offer to your users in a few different ways. However,
+this creates a truly unique problem—media files can be very, *very* large. So
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/pwa-with-offline-streaming/index.md:24]: I feel the punctuation is actually correct here as it's more of a preference. Is there something in the handbook that says to use a colon over the em dash? We could definitely remove the double use of **very** though.

File: src/site/content/en/media/media-streaming-basics/index.md [hunk 14]
```diff
@@ -0,0 +1,165 @@
...
+in the `<video>` element to a manifest file.
+
+```html
+<video controls>
+  <source src="manifest.mpd" type="application/dash+xml">
+</video>
+```
+
+{% Aside 'caution' %}
+This is valid, but doesn't actually work. Browsers don't natively support
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/media-streaming-basics/index.md:74]: Valid as in you can do it without errors but the browser can't do anything with it. At some point in the future I imagine you would be able to directly add the manifest to the `src` of the source element.

File: src/site/content/en/media/media-frameworks/index.md [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+## Video.js
+
+According tho their website, [Video.js] is a web video player built from the
+ground up for an HTML5 world. It supports HTML5 video and modern streaming
+formats such as DASH and HLS, as well as YouTube, Vimeo, and even Flash
+(through plugins). It supports video playback on desktop and mobile devices
+and looks good everywhere with CSS-based Skins.
+
+There are a few ways to use Video.js, but the easiest is to use the free CDN
+version provided by Fastly. You can learn more on how to get the player set up
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/media-frameworks/index.md:103]: Probably. I'll look for the right link to add.

File: src/site/content/en/media/media-frameworks/index.md [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+
+JW Player supports streaming with MPEG-DASH (paid version only), Digital rights
+management (DRM) (with Vualto), interactive advertisement, customization of the
+interface, and embeds. There is a well documented API and SDK. However, if you
+are just looking for a quick and free way to host your media then embedding
+YouTube videos are typically going to be your best free option.
+
+## Video.js
+
+According tho their website, [Video.js] is a web video player built from the
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/media-frameworks/index.md:96]:
```suggestion
According to their website, [Video.js] is a web video player built from the
```

File: src/site/content/en/media/media-file-basics/index.md [hunk 17]
```diff
@@ -45,46 +62,69 @@ A _codec_, or coder/decoder, is a compression format for video or audio data. Th
...
+| [WebM] | [AV1], [VP9]* | [Vorbis], [Opus] |
+
 &#42; Indicates the preferred video codec.
 
-_Bitrate_ is the maximum number of bits used to encode one second of a stream.
-The more bits used to encode a second of stream, the higher the potential
-detail and fidelity.
+## Bitrate and resolution
+
+**_Bitrate_** is the maximum number of bits used to encode one second of a stream.
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/media-file-basics/index.md:105]:
```suggestion
**Bitrate** is the maximum number of bits used to encode one second of a stream.
```

File: src/site/content/en/media/media-file-basics/index.md [hunk 18]
```diff
@@ -45,46 +62,69 @@ A _codec_, or coder/decoder, is a compression format for video or audio data. Th
...
+The more bits used to encode a second of stream, the higher the potential detail
+and fidelity. We provide more information about this concept in [Bitrate](/bitrate/).
+
+**_Resolution_** is the amount of information in a single frame of video, given as
```
Comment (derekherman) [inline src/site/content/en/media/media-file-basics/index.md:109]:
```suggestion
**Resolution** is the amount of information in a single frame of video, given as
```

File: src/site/content/en/media/media-accessibility/index.md [hunk 19]
```diff
@@ -95,12 +104,76 @@ manuscript
...
+```text
+WEBVTT
+
+00:00:05.000 --> 00:00:10.000 line:0 position:20% size:60% align:start
+The first line of the subtitles.
+```
+
+Another handy feature is the ability to style cues using CSS. Perhaps you want
+to use a gray linear gradient as the background, with a foreground color of
+`papayawhip` for all captions and all bold text colored `peachpuff`.
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/media-accessibility/index.md:135]: That a great point. I'll circle back on this.

File: src/site/content/en/media/bitrate/index.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/bitrate/index.md:18]:
```suggestion
learned how to change the container (extension) and codec of a media file. In
```

File: src/site/content/en/media/add-media/index.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/add-media/index.md:48]:
```suggestion
There is a lot of ground to cover in this section, get started by learning
```

**Comment by derekherman** [comment]
@jpmedley really excellent work finding all those issues. Sorry for the delay in getting back to you. I've been on vacation last week and this week. However, I know you are also going to be off next week so I sat down today and committed the bulk of the suggestions that could be merged easily. I'll do another pass tomorrow to fix the dates, merge conflicts, and any of the remaining unresolved suggestions. Thank you for reading the changes with a critical eye. Feels like you've found all the typos 😄  👍

File: src/site/content/en/media/pwa-with-offline-streaming/index.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/pwa-with-offline-streaming/index.md:50]: I see it starts at 0

File: src/site/content/en/media/media-frameworks/index.md [hunk 20]
```diff
@@ -0,0 +1,141 @@
...
+In most situations, that are beyond the needs of a basic web page, you are going
+to want to use a media framework to serve your content. Unless you are prepared
+to develop and maintain a rich feature set like offline playback, streaming,
+analytics, picture-in-picture, preview thumbnails, embedding, and monetization
+such as fill rate optimization, Ad scheduling, or header bidding to name a few
+then you should probably offload that complexity to an existing solution.
+
+This is where media frameworks come in. They provide you with a set of features,
+and conditions in which you can use those features, and then you have to decide
+which framework is right for your project. In the next article we will discuss
```
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/media-frameworks/index.md:41]: The article this sentences is referencing is co-authored

**Comment by derekherman** [comment]
@jpmedley I believe all the comments have been resolved. When you're able could you give this another look. As well, the workflow run needs to be approved so the lint and test can run.

Additionally, how do we add author photos?

**Comment by rachelandrew** [comment]
@derekherman I'll leave this for Joe's approval but I've hit the button to run the workflows for you :)

File: src/site/content/en/media/pwa-with-offline-streaming/index.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (jpmedley) [reply src/site/content/en/media/pwa-with-offline-streaming/index.md]: This one is from the Chicago Manual of Style, 17th edition, which is the Google standard. 6.61: "Between independent clauses it functions much like a semicolon". 6.85 suggests that colons can sometimes be substituted, but that rule does not cover the case of independent clauses.

File: src/site/content/en/media/pwa-with-offline-streaming/index.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (jpmedley) [reply src/site/content/en/media/pwa-with-offline-streaming/index.md:243]: That was supposed to be '()', since this is a method.

**Comment by jpmedley** [comment]
There are only two open items left, then we're done.

File: src/site/content/en/media/pwa-with-offline-streaming/index.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (derekherman) [reply src/site/content/en/media/pwa-with-offline-streaming/index.md:243]: Ah that makes more sense 👍

**Comment by derekherman** [comment]
@jpmedley I've merged your change, thank you for the review. From my POV we need to add the author images, which I'm unsure on how to do that, and we also need to move the demo PWA repo to Google's GitHub and update the URLs. I have a call today with Rob Kochman so I'll bring that up. Otherwise, it appears like we're getting very close. Could you explain how to get author photos added?

**Comment by rkochman** [comment]
I'm still working on getting the kino repo moved over. In the meantime, it looks like a Googler has to add your author image to the CDN for it to work on web.dev. Send me one that's 384x384, and I can do it.

**Comment by MichaelSolati** [comment]
@derekherman we've made a big change to how we handle author data, would you like me to create a pr to your branch making the changes?

**Comment by derekherman** [comment]
@MichaelSolati sorry I missed your comment. I've update the PR.

**Comment by derekherman** [comment]
@rkochman has there been any movement on transferring the PWA repository?

**Review by jpmedley — APPROVED** [review]

**Comment by jpmedley** [comment]
Are we ready to take this out of draft status?

**Comment by derekherman** [comment]
@jpmedley the only thing left is for the PWA repo on GitHub to be transferred and then update the links in the markdown. We could take it out of draft but it still shouldn't be merged until that happens.

**Comment by jpmedley** [comment]
Let's leave it in draft. That will prevent accidental merge.

**Comment by derekherman** [comment]
@jpmedley this is now ready for review to be merged.

**Pull request merged**